quantmsio/core/feature.py

```
@@ -23,17 +26,21 @@
         self._automaton = get_ahocorasick(self._mods_map)
 
     def extract_psm_msg(self, chunksize=2000000, protein_str=None):
-        P = Psm(self.mztab_path)
-        pep_dict = P.extract_from_pep(chunksize=100000)
+        psm = Psm(self.mztab_path)
+        pep_dict = psm.extract_from_pep(chunksize=100000)
         map_dict = {}
-        for psm in P.iter_psm_table(chunksize, protein_str):
-            for key, df in psm.groupby(["reference_file_name", "peptidoform", "precursor_charge"]):
+        for psm in psm.iter_psm_table(chunksize, protein_str):
```

Ruff (0.8.2)

Loop control variable `psm` overrides iterable it iterates

(B020)